### PR TITLE
[31979] Allow filtering WPs within IFC Models module

### PR DESCRIPTION
**Pull request by HDinger**
### ToDo

**Functionality** 
- [x] Changing a filter:
  - [x] changes the list of Work Packages 
  - [x] changes the url params
- [x] A deep link can be send and the filters are correctly initialised
- [x] Going back in the Browser changes the table accordingly if there was a change in the filter
- [x] Add tests

**Optics** 
- [x] Position filter button and container correctly in/below toolbar 
- [x] Hide button in model only view

### CODE REVIEW

**Comment by HDinger** [comment]
Ticket: 
https://community.openproject.com/projects/bim/work_packages/31979/activity

File: frontend/src/app/modules/ifc_models/bcf/container/bcf-container.component.ts [hunk 1]
```diff
@@ -1,47 +1,67 @@
...
 import {I18nService} from "core-app/modules/common/i18n/i18n.service";
 import {PathHelperService} from "core-app/modules/common/path-helper/path-helper.service";
 import {CurrentProjectService} from "core-components/projects/current-project.service";
 import {WorkPackageTableConfigurationObject} from "core-components/wp-table/wp-table-configuration";
-import { StateService } from '@uirouter/core';
+import {StateService} from '@uirouter/core';
 import {GonService} from "core-app/modules/common/gon/gon.service";
+import {QueryParamListenerService} from "core-components/wp-query/query-param-listener.service";
+import {InjectField} from "core-app/helpers/angular/inject-field.decorator";
+import {WorkPackagesListService} from "core-components/wp-list/wp-list.service";
```
Comment (openprojectci) [inline frontend/src/app/modules/ifc_models/bcf/container/bcf-container.component.ts]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/2bca7ebe6a58776d5479934ae44cca26a6cab30c/frontend/src/app/modules/ifc_models/bcf/container/bcf-container.component.ts#L11"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/ifc_models/bcf/container/bcf-container.component.ts [hunk 2]
```diff
@@ -1,51 +1,75 @@
...
+import {UrlParamsHelperService} from "core-components/wp-query/url-params-helper";
```
Comment (openprojectci) [inline frontend/src/app/modules/ifc_models/bcf/container/bcf-container.component.ts]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/32cb6f9d1699cd95247272702bf1f363faa5bfd7/frontend/src/app/modules/ifc_models/bcf/container/bcf-container.component.ts#L12"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/ifc_models/bcf/container/bcf-container.component.ts [hunk 3]
```diff
@@ -1,51 +1,84 @@
...
+import {untilComponentDestroyed} from "ng2-rx-componentdestroyed";
```
Comment (openprojectci) [inline frontend/src/app/modules/ifc_models/bcf/container/bcf-container.component.ts:13]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/a9d4131a530a40a936703bf6d0dd8c4fd15af08c/frontend/src/app/modules/ifc_models/bcf/container/bcf-container.component.ts#L13"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

**Review by oliverguenther — APPROVED** [review]

**Pull request merged**